sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1027,7 +1030,7 @@ class Analyzer(
    * The HAVING clause could also used a grouping columns that is not presented in the SELECT.
    */
   object ResolveMissingReferences extends Rule[LogicalPlan] {
```

Comment:
The `Filter` case of this `ResolveMissingReferences` rule doesn't look making much sense to me. I've opened #17874 for it. Here I'd make necessary change to pass tests.